### PR TITLE
add commentbox functionality, including: validation, vuex for storing…

### PR DESCRIPTION
**Pull request by kterekh**
… comments and deleting

### CODE REVIEW

File: src/common/Button.vue [hunk 1]
```diff
@@ -18,10 +20,11 @@ import { btnColor, btnType } from '@/common/types'
 export default class Button extends Vue {
   @Prop({ required: true }) color!: btnColor
   @Prop({ required: true }) size!: string
-  @Prop({ required: true }) title!: string
+  @Prop({ required: false }) title!: string
   @Prop({ default: false }) isDisabled!: boolean
-  @Prop() action!: string
+  @Prop({ required: false }) action!: string
   @Prop({ default: 'button' }) type!: btnType
+  @Prop({ required: false }) iconClass!: string
```
Comment (silchencko) [inline src/common/Button.vue:27]: iconClass is not required. What if there is no iconClass? What will happen to `<i>`?

File: src/common/Comment.vue [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import { Component, Vue, Prop } from 'vue-property-decorator'
+
+@Component({})
+export default class Comment extends Vue {
+  @Prop() comment!: string
+  @Prop({ required: false }) iconClass!: string
+  @Prop({ required: false }) action!: string
+
+  get comments (): [string] {
+    return this.$store.state.b.comments
```
Comment (silchencko) [inline src/common/Comment.vue:22]: Please find more meaningful name for the `b` module

File: src/common/Comment.vue [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop } from 'vue-property-decorator'
+
+@Component({})
+export default class Comment extends Vue {
+  @Prop() comment!: string
+  @Prop({ required: false }) iconClass!: string
```
Comment (silchencko) [inline src/common/Comment.vue:18]: if iconClass is not required, what will happen to template?

File: src/common/TextArea.vue [hunk 4]
```diff
@@ -32,7 +32,7 @@ export default class Input extends Vue {
   // value = ''
   @Prop({ default: 'false' }) isDisabled!: boolean
   @Prop() v!: Validation
-
+  @Prop() comment!: string
```
Comment (silchencko) [inline src/common/TextArea.vue:35]: Maybe `label` instead of `comment`?

File: src/common/Button.vue [hunk 5]
```diff
@@ -4,7 +4,9 @@
     class="btn"
     :class="btnClass"
     :disabled="isDisabled"
-    v-on:click="action">{{
+    v-on:click="action">
+    <i class="far" :class="iconClass"></i>
```
Comment (silchencko) [inline src/common/Button.vue:8]: My suggestion: hide this element if `iconClass` is not defined

File: src/components/FormDemoComponent.vue [hunk 6]
```diff
@@ -8,6 +8,7 @@
     />
     <br>
     <TextArea
+      comment="Comment:"
```
Comment (silchencko) [inline src/components/FormDemoComponent.vue:11]: `label`?

File: src/common/Comment.vue [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+      <span v-on:click="() => deleteComment(comment)"><i class="far" :class="iconClass"></i></span>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop } from 'vue-property-decorator'
+
+@Component({})
+export default class Comment extends Vue {
```
Comment (silchencko) [inline src/common/Comment.vue:16]: My suggestion is to call this class `Comments`  if it has multiple comments

**Review by silchencko — CHANGES_REQUESTED** [review]
Looks good to me! good work with vuex!

File: src/common/Button.vue (hunk 1 above)
Comment (kterekh) [reply src/common/Button.vue:27]: Hmm, still will be rendered. How to avoid this?

File: src/common/Button.vue (hunk 5 above)
Comment (kterekh) [reply src/common/Button.vue:8]: Got it, v-if will help here.

**Review by silchencko — APPROVED** [review]

**Pull request merged**